src/acquisition/covidcast/csv_importer.py

```diff
@@ -88,11 +88,11 @@ def find_issue_specific_csv_files(scan_dir, glob=glob):
     logger = get_structured_logger('find_issue_specific_csv_files')
     for path in sorted(glob.glob(os.path.join(scan_dir, '*'))):
       issuedir_match = CsvImporter.PATTERN_ISSUE_DIR.match(path.lower())
-      if issuedir_match and os.path.isdir(path):
+      if issuedir_match:
```

Comment:
interesting -- what will happen if `path` is a text file called `issue_12345678`? will the `find_csv_files` call on line 96 handle that gracefully, or crash?

Reply:
I take a try on this approach by picking up the issue_12345 by a text filed called issue_12345, here is some code of my approach: 
```python
path='prefix/to/the/data/issue_20200408.csv' 
issuedir_match = CsvImporter.PATTERN_ISSUE_DIR.match(('.').join(path.split('.')[:-1]))
```
it will pass on the tests but will fail on integration tests. Because the change of form of issue_path will effect file load_sensors.py,  so it will need some extra changes if I want to use the text file form as issue_123456.csv.

Reply:
sorry, i meant that we want the following:
1. if someone mistakenly creates a file which matches `CsvImporter.PATTERN_ISSUE_DIR` but is _not_ a directory
2. and we try to load the directory containing that bad file as a batch import
3. then the code should ignore the bad file

This was originally accomplished using the `and os.path.isdir(path)` clause. Without that clause, will the code crash in the above scenario?

Reply:
Gotcha! I previously hadn't make dir path to a valid dir so the test won't pass. I already add back the check code and add one more test of the invalid dir path.  One more question, can I know why in the function` find_csv_files`  check the csv path(the path format) instead to check the dir path? Should I add this`os.path.isdir(path)` also in the function` find_csv_files`?

Reply:
both `find_issue_specific_csv_files` and `find_csv_files` check that the paths they produce match a regular expression because the regex is used to extract information from the path (issue, source, time_value, geo_type, and signal). 

`find_issue_specific_csv_files` checks that the paths it produces are directories, because we've fed it bad input before by mistake 😉 . It doesn't get run very often, and each run has to be configured by hand, which means it's easy to accidentally do it wrong.

We run `find_csv_files` 24 times a day, so if we ever _did_ accidentally feed it a filesystem that had directories with names that should have been normal files, or normal files with names that should have been directories, we would have fixed it in early summer 2020. All the things that feed `find_csv_files` are automated and less susceptible to human error.